.github/workflows/workflow.yml

```diff
@@ -52,3 +52,45 @@ jobs:
 
     - name: Build & Test
       run: npm run test
+
+  test-proxy:
```

Comment:
the test job should also use the local action.  that gives you confidence in a PR when the checks pass that your code changes.  I think you also want 3 jobs.  build-test (it builds and units pass), test (e2e test using ./ local actions) and test-proxy (same as previous but with a proxy).  See the setup-node workflow which did a good job at this.

Reply:
I wanted to keep the scope down to just adding proxy support rather than also rewriting how the action is tested. I dont have much of an opinion yet on what an effective test for this action looks like so I just ported the existing one to also run in a proxied environment

Reply:
Ok for the extra job.  But making the existing job do a dot slash so it's consistent to what you're adding is needed.  That at least makes the existing and your new one consistent.

Reply:
I agree since this PR is already checking in the dist/ which was the blocker for not running the local action (I think). That change was made

Ultimately, it may make sense to split it into 3, first which tests the action code (npm install & test), then the other two test the action itself, which doesnt necessarily need to have anything to do with npm. The existing cache canary i think just writes a text file through the cache. I wonder if it makes more sense to more that type of canary to be local to the action repo?